### PR TITLE
Parameters: variants, not mux [v2]

### PR DESCRIPTION
**Pull request by clebergnu**
This PR is a collection of architectural and terminology arrangements.

It starts by not using the "mux" or "multiplexer" terminology in places where the (abstract) varianter interface is meant to be.  Then, it cleans up the `AvocadoParams` interfaces.  Finally, it moves the tests that are related to the mux code to the `yaml_to_mux` optional plugin, a primer that should be followed on other optional plugins.

---

Changes from v1 (#2336):
 - Added a missing `__init__.py` file on tests directory
 - Adapt new location of `yaml` files used on the tests
 - Rebased

### CODE REVIEW

File: avocado/core/test.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def record_and_warn(*args, **kwargs):
         elif isinstance(params, tuple):
             params, mux_path = params[0], params[1]
         self.__params = parameters.AvocadoParams(params, self.name,
-                                                 mux_path)
+                                                 mux_path, self.__log.name)
```
Comment (apahim) [inline avocado/core/test.py]: ok

File: avocado/core/runner.py [hunk 2]
```diff
@@ -188,10 +188,9 @@ def _tick(self):
                 self.job.result_proxy.notify_progress(False)
                 self.job._result_events_dispatcher.map_method('test_progress',
                                                               False)
-                if msg['paused']:
-                    reason = msg['paused_msg']
-                    if reason:
-                        self.job.log.warning(reason)
+                paused_msg = msg['paused']
+                if paused_msg:
+                    self.job.log.warning(paused_msg)
```
Comment (apahim) [inline avocado/core/runner.py]: Shouldn't the message come from `msg['paused_msg']`?

File: avocado/core/test.py [hunk 3]
```diff
@@ -54,6 +54,14 @@
 #: one job)
 COMMON_TMPDIR_NAME = 'AVOCADO_TESTS_COMMON_TMPDIR'
 
+#: The list of test attributes that are used as the test state, which
+#: is given to the test runner via the queue they share
+TEST_STATE_ATTRIBUTES = ('name', 'logdir', 'logfile',
+                         'status', 'running', 'paused',
+                         'time_start', 'time_elapsed', 'time_end',
+                         'fail_reason', 'fail_class', 'traceback',
+                         'params', 'timeout')
```
Comment (apahim) [inline avocado/core/test.py]: ok

File: selftests/unit/test_mux.py [hunk 4]
```diff
@@ -228,7 +228,7 @@ def test_fingerprint_order(self):
         variant1 = next(iter(mux1))
         variant2 = next(iter(mux2))
         self.assertNotEqual(variant1, variant2)
-        self.assertEqual(str(variant1), "{'mux_path': '', 'variant': "
+        self.assertEqual(str(variant1), "{'paths': '', 'variant': "
```
Comment (apahim) [inline selftests/unit/test_mux.py]: ok

File: optional_plugins/loader_yaml/avocado_loader_yaml/__init__.py [hunk 5]
```diff
@@ -101,8 +101,8 @@ def discover(self, reference, which_tests=loader.DEFAULT):
             return []
         mux_tree = mux.MuxTree(root)
         for variant in mux_tree:
-            params = parameters.AvocadoParams(variant, "YamlTestsuiteLoader",
-                                              ["/run/*"], output.LOG_UI.name)
+            params = parameters.AvocadoParams(variant, ["/run/*"],
+                                              output.LOG_UI.name)
```
Comment (apahim) [inline optional_plugins/loader_yaml/avocado_loader_yaml/__init__.py]: Yes, makes sense.

File: selftests/run [hunk 6]
```diff
@@ -26,6 +35,11 @@ def test_suite():
     for section in ('unit', 'functional', 'doc'):
         suite.addTests(loader.discover(start_dir=os.path.join(selftests_dir, section),
                                        top_level_dir=basedir))
+    # when other plugins get their own tests, make this a generic loop
+    if plugin_available('avocado-framework-plugin-varianter-yaml-to-mux'):
+        path = os.path.join(basedir, 'optional_plugins', 'varianter_yaml_to_mux')
+        suite.addTests(loader.discover(start_dir='tests',
+                                       top_level_dir=path))
```
Comment (apahim) [inline selftests/run]: ok, nice one.

**Comment by clebergnu** [comment]
It was intentional.  There's no need to have two keys, one for flagging and
one for holding the message.

**Comment by apahim** [comment]
@clebergnu, then you have to change the message generation: https://github.com/avocado-framework/avocado/blob/master/avocado/utils/process.py#L921

**Comment by clebergnu** [comment]
Closing in favor of #2349

**Pull request closed without merging**